src/sage/schemes/elliptic_curves/kummer_line.py

```diff
@@ -0,0 +1,1765 @@
+            ZQP, XQP = self.xDBL(XQ, ZQ)
+            return XQP, ZQP
+
+        A = XP + ZP
+        B = XP - ZP
+        C = XQ + ZQ
+        D = XQ - ZQ
+        DA = D * A
+        CB = C * B
+        t0 = DA + CB
+        t1 = t0 * t0
+        XQP = ZPQ * t1
+        t2 = DA - CB
+        t3 = t2 * t2
+        ZQP = XPQ * t3
+
+        return XQP, ZQP
+
+    def xDBLADD(self, XP, ZP, XQ, ZQ, XPQ, ZPQ):
+        r"""
+        Differential addition and doubling for `x`-only coordinate on Montgomery curve.
+
+        Cost: ``6M + 4S + 1m0 + 8a``
+
+        INPUT:
+
+        - ``XP``, ``ZP`` -- the coordinates of the Kummer point `x(P)`
+
+        - ``XQ``, ``ZQ`` -- the coordinates of the Kummer point `x(Q)`
+
+        - ``XPQ``, ``ZPQ`` -- the coordinates of the Kummer point `x(P-Q)`
+
+        OUTPUT: ``(X2P, Z2P)`` and ``(XQP, ZQP)``, the coordinates of `x(2P)` and `x(P+Q)` respectively
+
+        .. NOTE::
+
+            The formula does not behave well when `x(P-Q) = (1 : 0)` (should use `xDBL` instead)
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [0, 3, 0, 1, 0])
+            sage: K = KummerLine(E)
+            sage: P, Q = E(95, 54), E(15, 5)
+            sage: xP, xQ, xPQ = K(P), K(Q), K(P-Q)
+            sage: X2P, Z2P, XQP, ZQP = K.xDBLADD(*xP.XZ(), *xQ.XZ(), *xPQ.XZ())
+            sage: X2P/Z2P, XQP/ZQP
+            (25, 72)
+            sage: 2*P, P+Q
+            ((25 : 70 : 1), (72 : 27 : 1))
+        """
+        d = self._d
+
+        # In that case, P-Q = (0 : 0 : 1) = T on the curve, so P + Q = 2Q + T
+        # If 2Q = (X : Z), then 2Q + T = (Z : X)
+        if XPQ == 0:
+            X2P, Z2P = self.xDBL(XP, ZP)
+            ZQP, XQP = self.xDBL(XQ, ZQ)
+            return X2P, Z2P, XQP, ZQP
+
+        A = XP + ZP
+        AA = A * A
+        B = XP - ZP
+        BB = B * B
+        E = AA - BB
+        C = XQ + ZQ
+        D = XQ - ZQ
+        DA = D * A
+        CB = C * B
+        t0 = DA + CB
+        t1 = t0 * t0
+        XQP = ZPQ * t1
+        t2 = DA - CB
+        t3 = t2 * t2
+        ZQP = XPQ * t3
+        X2P = AA * BB
+        t4 = d * E
+        t5 = BB + t4
+        Z2P = E * t5
+
+        return X2P, Z2P, XQP, ZQP
+
+
+class KummerLinePoint(SageObject):
+    r"""
+    Kummer point on a Kummer line.
+
+    EXAMPLES::
+
+        sage: E = EllipticCurve(GF(101), [2, 3])
+        sage: K = KummerLine(E)
+        sage: P = E(95, 52)
+
+    A Kummer point can be constructed from `XZ`-coordinates::
+
+        sage: xP = KummerLinePoint(K, [95, 1]); xP
+        (95 : 1)
+
+    Or, it can be constructed from a point on the elliptic curve::
+
+        sage: xP = KummerLinePoint(K, P); xP
+        (95 : 1)
+
+    The point can also be created by calling the Kummer line::
+
+        sage: xP = K(P); xP
+        (95 : 1)
+    """
+
+    def __init__(self, parent, coords):
+        r"""
+        Create a point from the coordinates.
+
+        INPUT:
+
+            - ``parent`` - A Kummer line
```

Comment:
```suggestion
            - ``parent`` -- a Kummer line
```
Double dash for INPUT (special syntax for docs)